### PR TITLE
Add ConfigurationGeneralPage

### PR DESCRIPTION
**Pull request by drewnoakes**
Part of #7243

This "page" contains non-visible properties. It is present to support visibility conditions based on things like the target framework or language version.

As it stands, this PR will produce a strange looking page in the UI. A separate PR will be filed in CPS to filter not just non-visible properties, but also pages/categories containing only non-visible properties from the navigation tree.

A future PR in CPS will start to add visibility condition functions that will use the values provided via this new rule (by convention).

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7274)

### CODE REVIEW

**Review by swesonga — APPROVED** [review]

File: setup/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles/CommonFiles.swr [hunk 1]
```diff
@@ -23,6 +23,7 @@ folder "InstallDir:MSBuild\Microsoft\VisualStudio\Managed"
   file source="$(VisualStudioXamlRulesDir)Compile.FSharp.xaml"
   file source="$(VisualStudioXamlRulesDir)Compile.xaml"
   file source="$(VisualStudioXamlRulesDir)COMReference.xaml"
+  file source="$(VisualStudioXamlRulesDir)ConfigurationGeneralPage.xaml"
```
Comment (tmeschter) [inline setup/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles/CommonFiles.swr:26]: We already have a ConfigurationGeneral.xaml; we may want to give this a name that further differentiates it from that one.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ConfigurationGeneralPage.xaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <DataSource Persistence="ProjectFile"
+                SourceOfDefaultValue="AfterContext"
+                HasConfigurationCondition="True" />
+  </Rule.DataSource>
+
+  <StringProperty Name="TargetFrameworkIdentifier" Visible="False" ReadOnly="True" />
+
+  <StringProperty Name="TargetFrameworkVersion" Visible="False" ReadOnly="True" />
+  
+  <StringProperty Name="LangVersion" Visible="False" ReadOnly="True" />
```
Comment (tmeschter) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ConfigurationGeneralPage.xaml:25]: I would expect `TargetFrameworkIdentifier` and `TargetFrameworkVersion` to be present in more or less all .NET projects, but I'm not sure about `LangVersion`. We should at least make sure it is applicable to C#, VB, and F#. Also, if we're going to have an intrinsic visibility expression that uses this we should make sure that it works with the values returned by those languages.

**Review by tmeschter — APPROVED** [review]

File: setup/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles/CommonFiles.swr (hunk 1 above)
Comment (drewnoakes) [reply setup/Microsoft.VisualStudio.ProjectSystem.Managed.CommonFiles/CommonFiles.swr:26]: The similarity was intentional. Do you think it's confusing? In both CPS and the .NET Project System we have `ConfigurationGeneral` to represent general configured data about the project, not intended for display, which is similar to what this does (only this is a "page" for the UI).

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ConfigurationGeneralPage.xaml (hunk 2 above)
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/ConfigurationGeneralPage.xaml:25]: `LangVersion` doesn't appear to be available to VB or F#. We will need an additional means to identify the project's language in visibility conditions, I suspect.

**Pull request merged**